collector/lib/NetworkStatusNotifier.cpp

```diff
@@ -326,15 +335,27 @@ void NetworkStatusNotifier::AddConnections(::google::protobuf::RepeatedPtrField<
       // the case where we send a close event for a connection that it doesn't
       // know about.
       //
-      if (!rate_limiter.Allow(delta_entry.first.container())) {
-        CLOG_THROTTLED(INFO, std::chrono::seconds(5)) << "Rate limiting connections reported from container " << delta_entry.first.container();
+      std::string container = delta_entry.first.container();
+      if (!rate_limiter.Allow(container)) {
+        // [] operator will default construct the map element
+        // if it does not exist
+        rate_limited_containers[container]++;
         COUNTER_INC(CollectorStats::net_conn_rate_limited);
         continue;
       }
     }
 
+    added_events++;
     updates->AddAllocated(conn_proto);
   }
+
+  if (rate_limited_containers.size() != 0) {
+    for (auto container : rate_limited_containers) {
+      CLOG(INFO) << "Rate limited " << container.second << " connections from container " << container.first;
```

Comment:
```suggestion
    for (const auto& [id, events] : rate_limited_containers) {
      CLOG(INFO) << "Rate limited " << events << " connections from container " << id;
```

Reply:
The important part in this one is using `const auto&` to prevent more copies from being created, the binding just add a bit more readability IMO.

Reply:
Done